larsim/LegacyLArG4/ISCalculationCorrelated.cxx

```diff
@@ -0,0 +1,153 @@
+////////////////////////////////////////////////////////////////////////
+// \file  ISCalculationCorrelated.cxx
+// \brief Interface to algorithm class for calculating ionization electrons
+//        and scintillation photon based on simple microphysics arguments
+//        to establish anticorrelation between these two quantities.
+//
+//        To enable this in simulation, change LArG4Parameters variable
+//        in your fhicl file:
+//       
+//        services.LArG4Parameters.IonAndScintCalculator: "Correlated"
+//
+//        TO DO:
+//        [ ] Add fluctuations from Fano factor
+//        [ ] Get W_ph = 19.5eV from somewhere more centralized instead
+//            of hard-coding it into this algorithm
+//
+// \author  wforeman @ iit.edu
+////////////////////////////////////////////////////////////////////////
+
+#include "Geant4/G4LossTableManager.hh"
+#include "Geant4/G4EmSaturation.hh"
+#include "Geant4/G4ParticleTypes.hh"
+
+#include "larcore/CoreUtils/ServiceUtil.h"
+#include "larsim/LegacyLArG4/ISCalculationCorrelated.h"
+#include "lardata/DetectorInfoServices/LArPropertiesService.h"
+#include "lardata/DetectorInfoServices/DetectorPropertiesService.h"
+#include "larsim/Simulation/LArG4Parameters.h"
+#include "larsim/Simulation/LArVoxelCalculator.h"
+
+#include "messagefacility/MessageLogger/MessageLogger.h"
+#include "cetlib_except/exception.h"
+
+
+namespace larg4{
+
+  //----------------------------------------------------------------------------
+  ISCalculationCorrelated::ISCalculationCorrelated(CLHEP::HepRandomEngine&)
+  {
+  }
+
+  //----------------------------------------------------------------------------
+  ISCalculationCorrelated::~ISCalculationCorrelated()
+  {
+  }
+
+  //----------------------------------------------------------------------------
+  void ISCalculationCorrelated::Initialize()
```

Comment:
Suggest placing all initialization in the constructor instead of in a separate `Initialize` function.